// flex-layouts/lib/wrapper/Wrapper.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _elementum = require('elementum');

var _Container = require('./Container');

var _Container2 = _interopRequireDefault(_Container);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = _elementum.StyleSheet.createClassMap({
  'self': 'WrapperWrapper',
  'fillHeight': 'WrapperWrapper--FillHeight'
});

var Wrapper = function Wrapper(_ref) {
  var children = _ref.children;
  var _ref$fillHeight = _ref.fillHeight;
  var fillHeight = _ref$fillHeight === undefined ? true : _ref$fillHeight;
  return _react2.default.createElement(
    'div',
    { className: styles({ fillHeight: fillHeight }) },
    _react2.default.createElement(
      _Container2.default,
      null,
      children
    )
  );
};

Wrapper.propTypes = {
  children: _react.PropTypes.element
};

exports.default = Wrapper;
//ELEMENTUM_STYLES_COMMENT
var ELEMENTUM_EXTRACTED_STYLES = {
  '.WrapperWrapper': {
    position: 'relative',
    display: 'flex',
    width: '100%'
  },
  '.WrapperWrapper--FillHeight': {
    height: '100%'
  }
}; //ELEMENTUM_STYLES_COMMENT
