// flex-layouts/lib/column/Column.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _elementum = require('elementum');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = _elementum.StyleSheet.createClassMap({
  'self': 'ColumnColumn',
  'fill': 'ColumnColumn--Fill',
  'wrap': 'ColumnColumn--Wrap',
  'align=start': 'ColumnColumn--AlignStart',
  'align=center': 'ColumnColumn--AlignCenter',
  'align=end': 'ColumnColumn--AlignEnd',
  'justify=start': 'ColumnColumn--JustifyStart',
  'justify=center': 'ColumnColumn--JustifyCenter',
  'justify=end': 'ColumnColumn--JustifyEnd',
  'justify=spaceAround': 'ColumnColumn--JustifySpaceAround',
  'justify=spaceBetween': 'ColumnColumn--JustifySpaceBetween'
});

var Column = function Column(_ref) {
  var children = _ref.children;
  var fill = _ref.fill;
  var wrap = _ref.wrap;
  var align = _ref.align;
  var justify = _ref.justify;
  return _react2.default.createElement(
    'div',
    { className: styles({ fill: fill, wrap: wrap, align: align, justify: justify }) },
    children
  );
};

Column.propTypes = {
  children: _react.PropTypes.node,
  fill: _react.PropTypes.bool,
  align: _react.PropTypes.oneOf(['start', 'center', 'end']),
  justify: _react.PropTypes.oneOf(['start', 'center', 'end', 'spaceAround', 'spaceBetween'])
};

exports.default = Column;
//ELEMENTUM_STYLES_COMMENT
var ELEMENTUM_EXTRACTED_STYLES = {
  '.ColumnColumn': {
    display: 'flex',
    flexDirection: 'row',
    overflow: 'hidden',
    width: '100%'
  },
  '.ColumnColumn--Fill': {
    height: '100%'
  },
  '.ColumnColumn--Wrap': {
    flexWrap: 'wrap'
  },
  '.ColumnColumn--AlignStart': {
    alignItems: 'flex-start'
  },
  '.ColumnColumn--AlignCenter': {
    alignItems: 'center'
  },
  '.ColumnColumn--AlignEnd': {
    alignItems: 'flex-end'
  },
  '.ColumnColumn--JustifyStart': {
    justifyContent: 'flex-start'
  },
  '.ColumnColumn--JustifyCenter': {
    justifyContent: 'center'
  },
  '.ColumnColumn--JustifyEnd': {
    justifyContent: 'flex-end'
  },
  '.ColumnColumn--JustifySpaceAround': {
    justifyContent: 'space-around'
  },
  '.ColumnColumn--JustifySpaceBetween': {
    justifyContent: 'space-between'
  }
}; //ELEMENTUM_STYLES_COMMENT
