// flex-layouts/lib/layout/Layout.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _elementum = require('elementum');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = _elementum.StyleSheet.createClassMap({
  'self': 'LayoutLayout',
  'overflow': 'LayoutLayout--Overflow',
  'align=start': 'LayoutLayout--AlignStart',
  'align=center': 'LayoutLayout--AlignCenter',
  'align=end': 'LayoutLayout--AlignEnd',
  'justify=start': 'LayoutLayout--JustifyStart',
  'justify=center': 'LayoutLayout--JustifyCenter',
  'justify=end': 'LayoutLayout--JustifyEnd'
});

var Layout = function Layout(_ref) {
  var children = _ref.children;
  var _ref$grow = _ref.grow;
  var grow = _ref$grow === undefined ? 0 : _ref$grow;
  var _ref$shrink = _ref.shrink;
  var shrink = _ref$shrink === undefined ? 0 : _ref$shrink;
  var _ref$basis = _ref.basis;
  var basis = _ref$basis === undefined ? 'auto' : _ref$basis;
  var align = _ref.align;
  var justify = _ref.justify;
  var overflow = _ref.overflow;
  return _react2.default.createElement(
    'div',
    {
      className: styles({ align: align, justify: justify, overflow: overflow }),
      style: { flex: grow + ' ' + shrink + ' ' + basis }
    },
    children
  );
};

Layout.propTypes = {
  children: _react.PropTypes.oneOfType([_react.PropTypes.arrayOf(_react.PropTypes.element), _react.PropTypes.element]),
  grow: _react.PropTypes.number,
  shrink: _react.PropTypes.number,
  basis: _react.PropTypes.string,
  align: _react.PropTypes.oneOf(['start', 'center', 'end']),
  justify: _react.PropTypes.oneOf(['start', 'center', 'end']),
  overflow: _react.PropTypes.bool
};

exports.default = Layout;
//ELEMENTUM_STYLES_COMMENT
var ELEMENTUM_EXTRACTED_STYLES = {
  '.LayoutLayout': {
    display: 'flex',
    overflow: 'hidden',
    '-ms-overflow-style': '-ms-autohiding-scrollbar'
  },
  '.LayoutLayout--Overflow': {
    overflow: 'visible'
  },
  '.LayoutLayout--AlignStart': {
    alignItems: 'flex-start'
  },
  '.LayoutLayout--AlignCenter': {
    alignItems: 'center'
  },
  '.LayoutLayout--AlignEnd': {
    alignItems: 'flex-end'
  },
  '.LayoutLayout--JustifyStart': {
    justifyContent: 'flex-start'
  },
  '.LayoutLayout--JustifyCenter': {
    justifyContent: 'center'
  },
  '.LayoutLayout--JustifyEnd': {
    justifyContent: 'flex-end'
  }
}; //ELEMENTUM_STYLES_COMMENT
