// flex-layouts/lib/row/Row.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _elementum = require('elementum');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = _elementum.StyleSheet.createClassMap({
  'self': 'RowRow',
  'fill': 'RowRow--Fill',
  'wrap': 'RowRow--Wrap',
  'align=start': 'RowRow--AlignStart',
  'align=center': 'RowRow--AlignCenter',
  'align=end': 'RowRow--AlignEnd',
  'justify=start': 'RowRow--JustifyStart',
  'justify=center': 'RowRow--JustifyCenter',
  'justify=end': 'RowRow--JustifyEnd',
  'justify=spaceAround': 'RowRow--JustifySpaceAround',
  'justify=spaceBetween': 'RowRow--JustifySpaceBetween'
});

var Row = function Row(_ref) {
  var children = _ref.children;
  var fill = _ref.fill;
  var wrap = _ref.wrap;
  var align = _ref.align;
  var justify = _ref.justify;
  return _react2.default.createElement(
    'div',
    { className: styles({ fill: fill, wrap: wrap, align: align, justify: justify }) },
    children
  );
};

Row.propTypes = {
  children: _react.PropTypes.node,
  fill: _react.PropTypes.bool,
  align: _react.PropTypes.oneOf(['start', 'center', 'end']),
  justify: _react.PropTypes.oneOf(['start', 'center', 'end', 'spaceAround', 'spaceBetween'])
};

exports.default = Row;
//ELEMENTUM_STYLES_COMMENT
var ELEMENTUM_EXTRACTED_STYLES = {
  '.RowRow': {
    display: 'flex',
    flexDirection: 'column',
    overflow: 'hidden',
    width: '100%'
  },
  '.RowRow--Fill': {
    height: '100%'
  },
  '.RowRow--Wrap': {
    flexWrap: 'wrap'
  },
  '.RowRow--AlignStart': {
    alignItems: 'flex-start'
  },
  '.RowRow--AlignCenter': {
    alignItems: 'center'
  },
  '.RowRow--AlignEnd': {
    alignItems: 'flex-end'
  },
  '.RowRow--JustifyStart': {
    justifyContent: 'flex-start'
  },
  '.RowRow--JustifyCenter': {
    justifyContent: 'center'
  },
  '.RowRow--JustifyEnd': {
    justifyContent: 'flex-end'
  },
  '.RowRow--JustifySpaceAround': {
    justifyContent: 'space-around'
  },
  '.RowRow--JustifySpaceBetween': {
    justifyContent: 'space-between'
  }
}; //ELEMENTUM_STYLES_COMMENT
